tests/types/npm-test.json

```diff
@@ -209,16 +209,16 @@
       "description": "valid npm purl without version and with subpath, scope unencoded",
       "test_group": "advanced",
       "test_type": "roundtrip",
-      "input": "pkg:npm/%40babel/core#/googleapis/api/annotations/",
-      "expected_output": "pkg:npm/%40babel/core#googleapis/api/annotations",
+      "input": "pkg:npm/@babel/core#/googleapis/api/annotations/",
+      "expected_output": "pkg:npm/@babel/core#googleapis/api/annotations",
```

Comment:
:x: this change is wrong.
per PURL spec, the `@` is not a validcharacter in a PURL - it must be encoded.

https://github.com/package-url/purl-spec/blob/0a884d224637ae429733b0e871d4e3e4eae7689d/PURL-SPECIFICATION.rst?plain=1#L232-L242


having an `@` at this position will cause the follwoing parse result:
- subpath: `googleapis/api/annotations`
- version: `babel/core`
- type:`npm`
- no name 
- no namespace

Reply:
That's right, `@babel/core` could be mistaken for the version.

I reverted the original test that included the `@` character in the namespace.
https://github.com/package-url/purl-spec/blob/0a884d224637ae429733b0e871d4e3e4eae7689d/tests/types/npm-test.json#L212

We could set `expected_failure=true`, but I have observed that some PURL implementations support this particular case and it could cause problems in the tests.

The alternative is to eliminate these two tests.

Reply:
Thanks ;)